server/balancer.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"github.com/golang/protobuf/proto"
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+	"github.com/pingcap/kvproto/pkg/metapb"
+)
+
+// Balancer is an interface to select store regions for auto-balance.
+type Balancer interface {
+	Balance(cluster *raftCluster) (*BalanceOperator, error)
```

Comment:
I prefer argument. The interface seems clear and we can know easily that we will use the clusterInfo to do the balance. :-)

@qiuyesuifeng